### DISCUSSION TITLE
How to use external content libraries

### DISCUSSION THREAD

**Discussion by fperillo** [vmware/packer-examples-for-vsphere#791] [Help] [answered]
I have a content library on cluster A, let's say that its sphere is called clu_a.test.com. I did not crate it myself, I have little control on it.

Now I have packer installed in cluster B, and to reach the content library I go to the main menu, select "Content Library" item on the menu, and I'm shown 3 "dirs", ISO, OVA, TEMPLATE. 

The ISOs I need are in the ISO "dir", "Other types" tab.

I was not able to put these infos inside the configuration. I had to create a fake VM, assign the iso to the cd, wait for the file to be copied locally then set the config to point to the local copy... it means I have to browse the file system since esx mangles the file name and put it in a dir structure with lid as name.

How can I use a Content Library? How can I point to ISO\Other types\redhat....iso hosted on   clu_a.test.com?

**Comment by tenthirtyam** [comment]
As long as the two clusters are within the **same** vCenter Server instance you can modify your version to use the content library for the ISOs if they are stored there and not directly from the datastore.

`**iso_paths**` `([]string)` - A list of paths to ISO files in either a datastore or a content library that will be mounted to the VM.

Usage example (HCL):

```hcl
iso_paths = [
  "[nfs] iso/ubuntu-server-amd64.iso",
  "Packer/ubuntu-server-amd64/ubuntu-server-amd64.iso"
]
```

Two ISOs are referenced:

An ISO in the "iso" folder of the "nfs" datastore with the file name of "ubuntu-server-amd64.iso".
An ISO in the "Packer" content library with the item name of "ubuntu-server-amd64".

The plugin will check the path provided to determine if it's referring a content library for the ISO.

**Note**: All files in a content library have an associated item name. To determine the file name, view the datastore backing the content library or use the govc vSphere CLI.

The examples project is configured for this use case.

Essentially, you would need to modify the structure of `local.iso_path`s and have it reflect the content library name. Ideally you'd want a way to toggle the source from local datastore to content library as needed.

**Reply by fperillo** [reply, marked as answer]
Thank you for your reply.

Unfortunately the two clusters have different vCenter, so probably it explains why I was not able to use them.

I will ask to create a NFS share and load all the needed ISOs there. Or I will copy them to a local datastore...
